test/integration/scheduler/queue_test.go

```diff
@@ -587,26 +587,91 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			wantRequeuedPods:          sets.New("pod2"),
 			enableSchedulingQueueHint: []bool{true},
 		},
-
 		{
-			name:         "Pod rejected by the PodAffinity plugin is requeued when updating the label of the node to make it match the pod affinity",
-			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Label("node", "fake-node").Obj()},
+			name:         "Pod rejected by the interPodAffinity plugin is requeued when creating the node with the topologyKey label of the pod affinity",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node1").Label("aaa", "fake-node").Obj()},
 			pods: []*v1.Pod{
 				// - pod1 and pod2 will be rejected by the PodAffinity plugin.
 				st.MakePod().Name("pod1").PodAffinityExists("bbb", "zone", st.PodAffinityWithRequiredReq).Container("image").Obj(),
 				st.MakePod().Name("pod2").PodAffinityExists("ccc", "region", st.PodAffinityWithRequiredReq).Container("image").Obj(),
 			},
-
 			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
-				// Add label to the node which will make it match pod1's podAffinity.
-				if _, err := testCtx.ClientSet.CoreV1().Nodes().Update(testCtx.Ctx, st.MakeNode().Name("fake-node").Label("zone", "zone1").Obj(), metav1.UpdateOptions{}); err != nil {
-					return nil, fmt.Errorf("failed to update pod1: %w", err)
+				// Create the node which will make it match pod1's podAffinity.
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Create(testCtx.Ctx, st.MakeNode().Name("fake-node2").Label("zone", "zone1").Obj(), metav1.CreateOptions{}); err != nil {
+					return nil, fmt.Errorf("failed to create fake-node2: %w", err)
 				}
-				return map[framework.ClusterEvent]uint64{framework.NodeLabelChange: 1}, nil
+				return map[framework.ClusterEvent]uint64{framework.NodeAdd: 1}, nil
 			},
 			wantRequeuedPods:          sets.New("pod1"),
 			enableSchedulingQueueHint: []bool{true},
 		},
+		{
+			name: "Pod rejected by the interPodAffinity plugin is requeued when updating the node with the topologyKey label of the pod affinity",
+			initialNodes: []*v1.Node{
+				st.MakeNode().Name("fake-node").Obj(),
+			},
+			initialPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Label("affinity1", "affinity1").Container("image").Node("fake-node").Obj(),
+			},
+			pods: []*v1.Pod{
```

Comment:
I thought about it: if the original node has TopologyKey, it seems that it will be directly scheduled to this node. right?

Reply:
I guess a new Pod you tried hit [this case](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/interpodaffinity/filtering.go#L351-L355). To avoid the route, you should define the pod like:

```go
// This Pod doesn't have any label so that this PodAffinity doesn't match this Pod itself.
st.MakePod().Name("pod4").PodAffinityExists("affinity1", "region", st.PodAffinityWithRequiredReq).Container("image").Obj()
```

Reply:
I tried adding pod4, as you suggested, and added label in node ("region": "region-1") and when updating, ("region": "region-2")

```go
{
	name: "Pod rejected by the interPodAffinity plugin is requeued when updating the node with the topologyKey label of the pod affinity",
	initialNodes: []*v1.Node{
		st.MakeNode().Name("fake-node").Label("region", "region-1").Obj(),
	},
	initialPods: []*v1.Pod{
		st.MakePod().Name("pod1").Label("affinity1", "affinity1").Container("image").Node("fake-node").Obj(),
	},
	pods: []*v1.Pod{
		// - pod2, pod3, pod4 will be rejected by the PodAffinity plugin.
		st.MakePod().Name("pod2").Label("affinity1", "affinity1").PodAffinityExists("affinity1", "node", st.PodAffinityWithRequiredReq).Container("image").Obj(),
		st.MakePod().Name("pod3").Label("affinity1", "affinity1").PodAffinityExists("affinity1", "other", st.PodAffinityWithRequiredReq).Container("image").Obj(),
		// This Pod doesn't have any label so that this PodAffinity doesn't match this Pod itself.
		st.MakePod().Name("pod4").PodAffinityExists("affinity1", "region", st.PodAffinityWithRequiredReq).Container("image").Obj(),
	},
	triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
		// Add "node" label to the node which may make pod2 schedulable.
		// Update "region" label to the node which may make pod4 schedulable.
		if _, err := testCtx.ClientSet.CoreV1().Nodes().Update(testCtx.Ctx, st.MakeNode().Name("fake-node").Label("node", "fake-node").Label("region", "region-2").Obj(), metav1.UpdateOptions{}); err != nil {
			return nil, fmt.Errorf("failed to update fake-node: %w", err)
		}
		return map[framework.ClusterEvent]uint64{{Resource: framework.Node, ActionType: framework.UpdateNodeLabel}: 1}, nil
	},
	wantRequeuedPods:          sets.New("pod2", "pod4"),
	enableSchedulingQueueHint: []bool{true},
},
```

Reply:
But it still seems wrong. Can you help check it? We should modify the value of the region TopologyKey, right?

Reply:
@sanposhiho Can you help me take a look at this problem? :)

Reply:
Oops, looks like it's slipped from my list. Im busy on the urgent tasks for the code release and Kubecon, but surely will come back here.

Reply:
It should be enough to add a second node:
```go
		{
			name: "Pod rejected by the interPodAffinity plugin is requeued when updating the node with the topologyKey label of the pod affinity",
			initialNodes: []*v1.Node{
				st.MakeNode().Name("fake-node").Label("region", "region-1").Obj(),
				st.MakeNode().Name("fake-node2").Label("region", "region-2").Label("group", "b").Obj(),
			},
			initialPods: []*v1.Pod{
				st.MakePod().Name("pod1").Label("affinity1", "affinity1").Container("image").Node("fake-node").Obj(),
			},
			pods: []*v1.Pod{
				// - pod2, pod3, pod4 will be rejected by the PodAffinity plugin.
				st.MakePod().Name("pod2").Label("affinity1", "affinity1").PodAffinityExists("affinity1", "node", st.PodAffinityWithRequiredReq).Container("image").Obj(),
				st.MakePod().Name("pod3").Label("affinity1", "affinity1").PodAffinityExists("affinity1", "other", st.PodAffinityWithRequiredReq).Container("image").Obj(),
				// This Pod doesn't have any label so that this PodAffinity doesn't match this Pod itself.
				st.MakePod().Name("pod4").PodAffinityExists("affinity1", "region", st.PodAffinityWithRequiredReq).NodeAffinityIn("group", []string{"b"}, st.NodeSelectorTypeMatchExpressions).Container("image").Obj(),
				// TODO: We need to add one more pod here to test the pod is requeued when the value of topology label is updated.
				// more detail: https://github.com/kubernetes/kubernetes/pull/127470/files#r1806395330
			},
			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
				// Add "node" label to the node which may make pod2 schedulable.
				if _, err := testCtx.ClientSet.CoreV1().Nodes().Update(testCtx.Ctx, st.MakeNode().Name("fake-node").Label("node", "fake-node").Label("region", "region-2").Obj(), metav1.UpdateOptions{}); err != nil {
					return nil, fmt.Errorf("failed to update fake-node: %w", err)
				}
				return map[framework.ClusterEvent]uint64{{Resource: framework.Node, ActionType: framework.UpdateNodeLabel}: 1}, nil
			},
			wantRequeuedPods:          sets.New("pod2", "pod4"),
			enableSchedulingQueueHint: []bool{true},
		},
```

Pod4 can't be scheduled initially, because:
- `fake-node` doesn't have label `group` with value `b`, that is matching node affinity,
- `fake-node2` doesn't have any pod with `affinity1` in its region,

Updating `fake-node` merges regions of both nodes, so then `pod4` can be scheduled on `fake-node2`, as `fake-node` has a pod with `affinity1` and is in the same region as `fake-node2`.